src/Lumper.Lib/BSP/JsonContractResolver.cs

```diff
@@ -1,41 +1,45 @@
 namespace Lumper.Lib.BSP;
+
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using Lumper.Lib.BSP.IO;
-using Lumper.Lib.BSP.Lumps;
+using IO;
+using Lumps;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Serialization;
 
-public class JsonContractResolver(bool sortProperties, bool ignoreOffset) : DefaultContractResolver
+public class JsonContractResolver : DefaultContractResolver
 {
-    public bool SortProperties { get; } = sortProperties;
-    public bool IgnoreOffset { get; } = ignoreOffset;
+    public required bool SortProperties { get; init; }
+    public required bool IgnoreOffset { get; init; }
```

Comment:
Benefit of this is that you have to explicitly name the arguments, which is nice in the case of two bools. Also, this seems to be the convention mostly libraries follow nowadays. We instantiate the class in `BspFile.cs` here:
```c#
 var jsonSerializerSettings = new JsonSerializerSettings {
            ContractResolver = new JsonContractResolver {
                SortProperties = sortProperties, IgnoreOffset = ignoreOffset
            },
            Formatting = Formatting.Indented
        };
```
Isn't the initializer-based approach preferable to `new JsonContractResolver(true, true)` here?

Also, whilst I'm maybe overzealous with some of the new language features (perhaps e.g. primary ctors), object initializers are very standard part of C# now, the only new-ish thing is `required` which takes 30s to look up if you don't know already.

Reply:
The thing is you *have to* do it like this now. The `JsonContractResolver(true, true)` is a bad example because it would be `JsonContractResolver(sortProperties, ignoreOffset)` instead of `JsonContractResolver {  SortProperties = sortProperties, IgnoreOffset = ignoreOffset }`. Even without variables you could still do `JsonContractResolver(SortProperties: true, IgnoreOffset: true)` to make it clear what the two bools are. So no, I don't think this is beneficial here.

Reply:
Yes you can do it with named params, but when unsure I'm generally following what existing .NET libraries do, which is use object initializers. We could go back and forth over and over on using new language features, but if you want to avoid them you need to give more substantial objections to them than "it's nicer" or "it's not obvious how it works".

Reply:
Using ctor now